lib/Common/Memory/RecyclerPageAllocator.cpp

```diff
@@ -25,6 +25,24 @@ bool RecyclerPageAllocator::IsMemProtectMode()
     return recycler->IsMemProtectMode();
 }
 
+char * RecyclerPageAllocator::TryAllocFromZeroPages(uint pageCount, PageSegmentBase<VirtualAllocWrapper> ** pageSegment, PageHeapMode pageHeapFlags)
+{
+    char* pages = nullptr;
+
+    pages = TryAllocFromZeroPagesList(pageCount, pageSegment, this->zeroPageQueue.pendingZeroPageList);
+    if (pages != nullptr)
+    {
+        // zero out
+        memset(pages, 0, pageCount*AutoSystemInfo::PageSize);
+        return pages;
+    }
+
+    pages = TryAllocFromZeroPagesList(pageCount, pageSegment, this->zeroPageQueue.freePageList);
```

Comment:
yeah, I hesitated here. it was freePages first. Then I thought pendingZeroPage might have bigger chance to hit. But I think you are right, checking existence of freePages cost should be very low comparing to zero out total 4 pages. let me change back